test/integration/output_test.py

```diff
@@ -252,3 +252,69 @@ def test_mcoe_filled(fast_out_filled, df_name, expected_nuke_fraction, tolerance
         fast_out_filled.__getattribute__(df_name)()
     )
     assert abs(actual_nuke_fraction - expected_nuke_fraction) <= tolerance
+
+
+@pytest.mark.parametrize(
+    "variation, check",
+    [
+        ("test_local", "df_equal"),
+        ("test_invalid_engine_url", "df_equal"),
+        ("test_local", "same_keys"),
+        ("test_invalid_engine_url", "same_keys"),
+        ("test_local", "valid_db"),
+        ("test_invalid_engine_url", "valid_db"),
+    ],
+)
+def test_pudltabl_pickle(
+    fast_out_annual, pudl_settings_fixture, monkeypatch, variation, check
+):
+    """Test that PudlTabl is serializable with pickle.
+
+    Because pickle is insecure, bandit must be quieted for some lines of this test. This
+    test attempts to simulate the situation where PudlTabl is restored in the same
+    environment that created it 'test_local' and in a different environment from the one
+    that created it 'test_invalid_engine_url'.
+    """
+    import pickle  # nosec
+    from copy import copy
+    from io import BytesIO
+
+    # need to monkeypatch `get_defaults` because it is used in PudlTabl.__setstate__
+    # and the real one does not work in GitHub actions because the settings file it
+    # uses does not exist
+    monkeypatch.setattr(
+        "pudl.workspace.setup.get_defaults", lambda: pudl_settings_fixture
+    )
+
+    # make a copy so changes here don't affect other tests
+    pudltabl = copy(fast_out_annual)
```

Comment:
Another alternative would be to change the scope of `fast_out_annual` back to `function` (the default) - then we don't have to worry about inter-test entanglement. I just ran the tests both ways and neither seems slower than the other - what do you think?

Reply:
Agreed, that is better.